### ISSUE TITLE
Feature Request: coloured and smaller guides for noise reduction.

### ISSUE THREAD

**Issue by marcdraco** [MrKepzie/Natron#1539]
This one should be easy to do but it would make life somewhat easier for hacks like me.

When I fire up a noise reduction node the sample frame (which is white) appears around the whole frame. Now this might be obvious, but coming from NEAT Video it was a little odd. Reason being it that most of the frame is detail and not noise even though the noise (primarily chroma noise) is overlayed on the whole thing.

Unless I'm mistaken, noise by nature is the same across the whole frame (averaged out) so you only need a sample of the noise taken from a section where there is little or no luminance shift - detail in other words.

If the sample area could be made smaller and more obvious - i.e. a square, perhaps coloured green or red to show it up, it would be easier to know where the devil it is the first time you use the node AND if coloured, it would be easier to find when you're moving it about a comp looking for a suitable area.

**Comment by devernay** [comment]
Actually the method we implemented should also work on the whole image in most cases, although it works better if only flat areas are selected

**Comment by marcdraco** [comment]
I agree- in fact, this problem only came to light when I found myself faced with a large area with rolling details - a stage curtain. While the noise reduction was pretty damn good, it was a difficult balanced between losing noise and losing detail in the curtain folds.

Honestly, it's a minor issue for the less technically minded like me. :)

I've added a feature request in relation to this to aid selecting areas by average brightness (the more I think on, perhaps averaged over several frames).

It appears from my tests that my sensor (Canon) is producing most noise in areas where there is less light. Even if it can detect detail in those areas. This is, I guess, because there are fewer photons for the pit to detect leading to random noise. As there are more photons (brighter areas) the problem almost vanishes - even though there is still plenty of detail.

Hence, if we can apply some form of mask based on brightness, we could tell the denoiser where we need to apply the strongest effects.

I hope that makes sense and again, thank you for all the work you do for us.
